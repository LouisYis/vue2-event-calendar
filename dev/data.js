const date = new Date()
const year = date.getFullYear()
const month = date.getMonth() + 1
export default [
  {
    date: `${year}-${month}-8`,
    title: 'buy something'
  },
  {
    date: `${year}-${month + 1}-2`,
    title: 'remember homework'
  },
  {
    date: `${year}-${month + 1}-15`,
    title: 'music festival'
  },
  {
    date: `${year}-${month + 2}-6`,
    title: 'a course of lectures'
  }
]
