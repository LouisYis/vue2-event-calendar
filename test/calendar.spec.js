import { mount, shallow } from '@vue/test-utils';
import Calendar from '../src/index';
import data from '../dev/data';

describe('Calendar component', () => {
  test('component snapshot', () => {
    const wrapper = mount(Calendar, {
      propsData: {
        value: new Date('2018-01-01'),
        dateData: data
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test('mode is work', () => {
    const wrapper = mount(Calendar, {
      propsData: {
        value: new Date('2018-01-01')
      }
    });

    const weekWrapper = mount(Calendar, {
      propsData: {
        mode: 'week',
        value: new Date('2018-01-01')
      }
    });

    const monthRow = wrapper.findAll('.vue-calendar-body-row');
    const weekRow = weekWrapper.findAll('.vue-calendar-body-row');

    expect(monthRow.length).toBe(6);
    expect(weekRow.length).toBe(1);
  });

  test('has require props', () => {
    const { value, dateData, onMonthChange, mode, prefixCls } = Calendar.props;
    expect(value).not.toBe(undefined);
    expect(dateData).not.toBe(undefined);
    expect(onMonthChange.type).toBe(Function);
    expect(mode).not.toBe(undefined);
    expect(prefixCls.default).toBe('vue-calendar');
  });

  test('has currect change action', () => {
    const prefixCls = 'kit-calendar';
    const wrapper = shallow(Calendar, {
      propsData: {
        value: new Date('2018-01-01'),
        prefixCls
      }
    });

    const prev = wrapper.find(`.${prefixCls}-prev`);
    const next = wrapper.find(`.${prefixCls}-next`);
    const title = wrapper.find(`.${prefixCls}-header-date`);

    next.trigger('click');
    wrapper.vm.$nextTick(() => {
      expect(title.text()).toBe('2018-02');
    });

    prev.trigger('click');
    wrapper.vm.$nextTick(() => {
      expect(title.text()).toBe('2018-01');
    });
  });
});
