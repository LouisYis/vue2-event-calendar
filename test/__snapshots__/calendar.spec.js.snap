// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar component component should match snapshot 1`] = `
<div
  class="vue-calendar is-month"
>
  <div
    class="vue-calendar-header"
  >
    <div
      class="vue-calendar-header-left"
    />
    <div
      class="vue-calendar-header-center"
    >
      <a
        class="vue-calendar-control vue-calendar-prev"
      >
        &lt;
      </a>
      <span
        class="vue-calendar-header-date"
      >
        2018-01
      </span>
      <a
        class="vue-calendar-control vue-calendar-next"
      >
        &gt;
      </a>
    </div>
    <div
      class="vue-calendar-header-right"
    />
  </div>
  <div
    class="vue-calendar-week-title"
  >
    <div
      class="vue-calendar-week-title-item"
    >
      Sun
    </div>
    <div
      class="vue-calendar-week-title-item"
    >
      Mon
    </div>
    <div
      class="vue-calendar-week-title-item"
    >
      Tue
    </div>
    <div
      class="vue-calendar-week-title-item"
    >
      Wed
    </div>
    <div
      class="vue-calendar-week-title-item"
    >
      Thu
    </div>
    <div
      class="vue-calendar-week-title-item"
    >
      Fri
    </div>
    <div
      class="vue-calendar-week-title-item"
    >
      Sat
    </div>
  </div>
  <div
    class="vue-calendar-body"
  >
    <div
      class="vue-calendar-body-row"
    >
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
    </div>
    <div
      class="vue-calendar-body-row"
    >
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
    </div>
    <div
      class="vue-calendar-body-row"
    >
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
    </div>
    <div
      class="vue-calendar-body-row"
    >
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
    </div>
    <div
      class="vue-calendar-body-row"
    >
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
    </div>
    <div
      class="vue-calendar-body-row"
    >
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
      <div
        class="vue-calendar-day-item"
      />
    </div>
  </div>
</div>
`;
